### ISSUE TITLE
Any fix for date.tomorrow?

### ISSUE THREAD

**Issue by jjaimealeman** [SilentVoid13/Templater#1024]
**Plugin information (please complete the following information):**

-   OS: EndeavourOS (arch based distro)
-   Templater version: v.1.16.0
-   Obsidian version: v.1.1.9
-   Templater settings: folder location = `/templater`

**Describe the bug**
From the examples page:
`<% tp.file.creation_date("dddd Do MMMM YYYY HH:mm:ss") %> Created` works fine 👍
`<% tp.file.last_modified_date("dddd Do MMMM YYYY HH:mm:ss") %> Modified` does not 👎
![image](https://user-images.githubusercontent.com/1428292/217124163-6831c5be-92d5-46e4-bea6-a93caad68422.png)

**Expected behavior**
I create new file, insert template. Then add content and close file.
When I reopen file and made edits, close, reopen, the `last_modified_date` time stamp does not change. It shows the exact same time stamp as `creation_date`.

**ADDITIONAL NOTES**
All the examples for inserting today and yesterday work fine 👍
But the example for tomorrow display only todays date, not tomorrow 👎

--- 

I was sent here by a Discord user after I had asked for a method to insert a template into my Daily Notes that simple link `yesterday | today | tomorrow`.

I like all the advanced features of this template and seems to have lots of advanced possibilities. I'm just trying to get the dates to format correctly before determining if I should keep this plugin.

**Comment by welpdx** [comment]
### Issue 1. `tp.file.last_modified_date()` is the same as created date.

This is because there is caching going on with obsidian. You can fix this by pressing F5 in console or in command palate -> `Reload app without saving`. This will refresh that cache. You could also close and open that tab again. 

If that doesn't work, try this:
```js
moment(this.app.workspace.getActiveFile().stat.mtime).format("dddd Do MMMM YYYY HH:mm:ss")
```

I created a new vault and was able to replicate that issue. But after Reloading I somehow never had that same problem again. 🤷


### Issue 2. But the example for tomorrow display only todays date, not tomorrow
Are you referring to this?
```text
File's title date + 1 day (tomorrow): <% tp.date.now("YYYY-MM-DD", 1, tp.file.title, "YYYY-MM-DD") %>
```

According to the doc, this example is for the use case when using daily notes and the `tp.file.title` is a date. 
Let me explain that above code:


| Arguments        | Doc example   | Explained in plain english                                                                                | My example                                                                                                                                                                    |
|------------------|---------------|-----------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| format           | "YYYY-MM-DD"  | The final date format you want                                                                            | YYMMDD                                                                                                                                                                        |
| offset           | 1             | offset, add one day to today's date                                                                       | 1                                                                                                                                                                             |
| reference        | tp.file.title | a string, in this case the string from the title                                                          | "2023-02-02"                                                                                                                                                                  |
| reference_format | "YYYY-MM-DD"  | what is the format of the above string? So that moment knows and convert   that string to a moment object | YYYY-MM-DD                                                                                                                                                                    |
|                  |               |                                                                                                           | This will take the string, convert it using the   `reference_format`, then add one day to the date, then output it as   "YYMMDD" format. The final output should be "230203". |

If the title does not contain a date, then do not provide tp.date.now() with the last 2 arguments. i.e., `<% tp.date.now("YYYY-MM-DD", 1) %>`. If the last 2 arguments are provided, instead of getting the time now, it will get the date mentioned in the `reference`.

<details>
           <summary style="font-size:20">demonstration</summary>
           <img src="https://i.imgur.com/6mycull.gif" alt="demo" />
</details>

**Comment by jjaimealeman** [comment]
@welpdx 

Thanks so much for the detailed reply.
I will try your suggestions!
👍🏽

**Comment by Zachatoo** [comment]
Has this issue been resolved for you @jjaimealeman ?

**Comment by jjaimealeman** [comment]
> Has this issue been resolved for you @jjaimealeman ?

I came up with my own solution.

A `cronjob` that creates the Daily Note with a link to Yesterday & Tomorrow. I now have all 670+ Daily Notes linked 👍 

I've not revisited this plugin since my initial post.

Thank you though for following up amigo. It is appreciated.
